tcpip/stack/stack.go

```diff
@@ -1040,29 +1040,32 @@ func (s *Stack) LeaveGroup(protocol tcpip.NetworkProtocolNumber, nicID tcpip.NIC
 	return s.RemoveAddress(nicID, multicastAddr)
 }
 
+// Close the stack. This wipes cached information (i.e. route tables), so many stack functions will
+// return an error once a stack is closed.
 func (s *Stack) Close() error {
-	s.mu.RLock()
-	defer s.mu.RUnlock()
+	s.mu.Lock()
+	defer s.mu.Unlock()
```

Comment:
The data race occurred because we modified s.nics, but only held a read lock.  Grabbing a write lock instead fixes the issue.